File: Demo/JCTiledViewDemo/DetailView.m
//
//  DetailView.m
//  JCTiledViewDemo
//
//  Created by Jesse Collis on 8/02/12.
//  Copyright 2012 JC Multimedia Design. All rights reserved.
//

#import "DetailView.h"

@implementation DetailView

- (id)initWithFrame:(CGRect)frame
{
  if ((self = [super initWithFrame:frame]))
  {
    self.backgroundColor = [UIColor blackColor];
    
    _textLabel = [[UILabel alloc] initWithFrame:[self bounds]];
    self.textLabel.font = [UIFont systemFontOfSize:17.0f];
    self.textLabel.textColor = [UIColor whiteColor];
    self.textLabel.textAlignment = UITextAlignmentLeft;
    self.textLabel.backgroundColor = [UIColor clearColor];
    
    [self addSubview:self.textLabel];
  }
  return self;
}

- (CGSize)sizeThatFits:(CGSize)size
{
  return CGSizeMake(size.width, 20.f);
}

- (void)layoutSubviews
{
  self.textLabel.frame = CGRectInset(self.bounds, 2.f, 2.f);
}

@end

File: Demo/JCTiledViewDemo/DetailView.swift
//
//  DetailView.swift
//  JCTiledViewDemo
//
//  Created by Jesse Collis on 28/9/17.
//  Copyright © 2017 JC Multimedia Design. All rights reserved.
//

import UIKit

class DetailView: UIView {
  
  var textLabel: UILabel?
  
  override init(frame: CGRect) {
    
    super.init(frame: frame)

    setupTextLabel()
  }
  
  required init?(coder aDecoder: NSCoder) {
    super.init(coder: aDecoder)
    
    setupTextLabel()
  }
  
  private func setupTextLabel() {
    let textLabel = UILabel()
    textLabel.textColor = .white
    textLabel.backgroundColor = .clear
    textLabel.translatesAutoresizingMaskIntoConstraints = false
    self.textLabel = textLabel
    
    addSubview(textLabel)
    
    textLabel.activateSuperviewHuggingConstraints(UIEdgeInsetsMake(2, 2, 2, 2))
  }
}
